rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -0,0 +1,126 @@
+        template_vars: Optional[Dict] = None,
+        use_e2e: bool = False,
+        source_name: Text = None,
+    ):
+        super().__init__(interpreter, domain, template_vars, use_e2e, source_name)
+
+    async def read_from_file(self, filename: Text) -> List[StoryStep]:
+        try:
+            file_content = rasa.utils.io.read_file(filename)
+            yaml_content = rasa.utils.io.read_yaml(file_content)
+
+            for key, value in yaml_content.items():
+                if key == KEY_STORIES:
+                    self._parse_stories(value)
+                    self._add_current_stories_to_result()
+                    return self.story_steps
+                else:
+                    logger.warning(f"Unexpected key {key} found in {self.source_name}")
+
+        except ValueError:
+            logger.error(f"Failed to read {filename}, it will be skipped")
+
+        return []
+
+    def _parse_stories(self, stories: List[Dict[Text, Any]]) -> None:
+
+        for story_item in stories:
+            if not isinstance(story_item, dict):
+                logger.warning(
+                    f"Unexpected block found in {self.source_name}: \n"
+                    f"{story_item}\n"
+                    f"Items under the `{KEY_STORIES}` key must be YAML dictionaries."
+                    f"It will be skipped."
+                )
+                continue
+
+            if KEY_PLAIN_STORY in story_item.keys():
+                self._parse_plain_story(story_item)
+
+    def _parse_plain_story(self, story_item: Dict[Text, Any]) -> None:
+        story_name = story_item.get(KEY_PLAIN_STORY, "")
+        steps = story_item.get(KEY_PLAIN_STORY_STEPS, "")
+
+        self._new_story_part(story_name, self.source_name)
+
+        for step in steps:
+            self._parse_step(step)
+
+    def _parse_step(self, step: Dict[Text, Any]):
+
+        if KEY_PLAIN_STORY_USER_UTTERANCE in step.keys():
+            self._parse_plain_story_user_utterance(step)
+        if KEY_SLOT_NAME in step.keys():
+            self._parse_slot(step)
+        if KEY_ACTION in step.keys():
+            self._parse_action(step)
+
+    def _parse_plain_story_user_utterance(self, step: Dict[Text, Any]):
+
+        user_utterance = step.get(KEY_PLAIN_STORY_USER_UTTERANCE, "").strip()
+        entities = step.get(KEY_PLAIN_STORY_ENTITIES, "")
+
+        if not user_utterance.startswith(INTENT_MESSAGE_PREFIX):
+            logger.warning(
+                f'User intent "{user_utterance}" should start with '
+                f'"{INTENT_MESSAGE_PREFIX}"'
+            )
+        else:
+            user_utterance = user_utterance[1:]
+
+        entities_list = []
+        for entity in entities:
+            if isinstance(entity, dict):
+                for key, value in entity.items():
+                    entities_list.append({"entity": key, "value": value})
+            else:
+                entities_list.append({"entity": entity, "value": ""})
```

Comment:
I'm not sure I understand that. The idea is to always have the `value` key, even if it's not specified.
```yaml
- user: /enter_data
  entities:
  - name
```

Reply:
Ok, but what's the use case for not specifying the entity value? And rather than using an empty string, should we just use the entity name as value in that case?